features/admin.applications.v1/components/settings/access-configuration.tsx

```diff
@@ -1069,7 +1069,9 @@ export const AccessConfiguration: FunctionComponent<AccessConfigurationPropsInte
             );
         }
 
-        if (applicationTemplateId === ApplicationTemplateIdTypes.M2M_APPLICATION) {
+        if (applicationTemplateId === ApplicationTemplateIdTypes.M2M_APPLICATION ||
+            template["originalTemplateId"] === ApplicationTemplateIdTypes.MCP_CLIENT_APPLICATION
```

Comment:
Better if we can have an issue to track this limitation

Reply:
`Due to the urgency, we are using conditional rendering to show/hide authentication configurations in the MCP template. Ideally the application edit component should be improved to render the fields conditionally according to the template obtained from the application template API. This will be addressed in a follow up effort.`

Reply:
Created an issue.
https://github.com/wso2/product-is/issues/23681